tests/integration/testlib/assertlib.py

```diff
@@ -53,3 +54,17 @@ def assert_absent(*ifnames):
 
     current_state = statelib.show_only(ifnames)
     assert not current_state[INTERFACES]
+
+
+def iface_macs(state):
+    for ifstate in state[Interface.KEY]:
+        yield ifstate[Interface.MAC]
+
+
+def assertMacAddress(state, expected_mac=None):
+    """ Asserts that all MAC addresses of ifaces in a state are the same """
+    macs = list(iface_macs(state))
+    if expected_mac:
+        assert set(macs) == {expected_mac}
```

Comment:
Thanks for letting me know that `type({'abc'}) == set`.

On second through, other people might have the same wonder. 
To save that, change it to `set((expected_mac, ))` might help others understand.